### ISSUE TITLE
Markdown filter does not use extensions

### ISSUE THREAD

**Issue by matthiasrohmer** [grow/grow#874]
Currently the markdown filter does directly pass any input to `markdown.markdown()` which has not loaded any of the extensions that may be configured in the `podspec.yaml`. Therefore when rendering content from a data source like a JSON or YAML file something like `{{ '[sourcecode:html]\n' + section.preview + '\n[/sourcecode]')|markdown|safe }}` is not possible.

The changes introduced in #867 allow for a dirty workaround as soon as they are released: 
```jinja2
{% set md, html = doc.format._markdown %}
{{ md.convert('[sourcecode:html]\n' + section.preview + '\n[/sourcecode]')|safe }}
```

Maybe it would make sense to move the whole creation of a markdown instance to pod-level instead of crafting it everytime a `MarkdownDocumentFormat` is created and then also use this instance in the markdown filter?

**Comment by Zoramite** [comment]
I've reworked the way that the extensions are loaded to not be inside of the document format directly. I agree that is not a good place for it. The filter should now load the extensions the same way that the normal markdown formatted documents would.

We can't do a pod wide cached instance of markdown because of the extensions. If we did a cached extension things like the TOC wouldn't work correctly.
